container.go

```diff
@@ -348,3 +348,17 @@ func getPsEnvVars(checkpointOutputDir string, pid uint32) (envVars []string, err
 
 	return
 }
+
+// findPsTreeByPID performs a short-circuiting depth-first search to find
+// a process with a given PID in a process tree.
+func findPsTreeByPID(root *crit.PsTree, targetPID uint32) *crit.PsTree {
+	if root.PID == targetPID {
+		return root
+	}
+	for _, child := range root.Children {
+		if foundProcess := findPsTreeByPID(child, targetPID); foundProcess != nil {
```

Comment:
🤔 I think it is safe to name this variable as just `process`, since the condition implies that it is the found process if it is not nil. `foundProcess` sounds like a boolean to me.